### DISCUSSION TITLE
v0.6.0

### DISCUSSION THREAD

**Discussion by mattdangerw** [keras-team/keras-hub#1130] [General]
KerasNLP is adding experimental support for Jax and PyTorch backends on top of the **Keras Core** library. Read the [anouncement](https://keras.io/keras_core/announcement/), and browse the [full library documentation](http://keras.io/keras_core/), including how to specify the backend when running your code.

Support for both Jax and PyTorch is still experimental, expect some rough edges and please give us feedback!

## Summary
* This release should be equivalent to `0.5.2` with the addition of multi-backend support.
* The following API symbols are currently restricted to the tensorflow backend:
  * `keras_nlp.models.DebertaV3*`
  * `keras_nlp.models.FNet*`
  * `keras_nlp.metrics`
  * `keras_nlp.samplers.BeamSampler`
  * `keras_nlp.samplers.ContrastiveSampler`
* Note that there are two ways you can run on top of Tensorflow.
  * If you run your scripts/colab without any changes, KerasNLP will use **tf.keras** for all layer and modeling implementations. This should be a no-op from previous releases of the library.
  * If you run your scripts/colab with `KERAS_BACKEND={jax, torch, tensorflow}`, you will be trying the new **Keras Core** library, using the specified backend. This is a great way to test out the future of the library!
  * Full details on runtime specification is available along with the [Keras Core documentation](http://keras.io/keras_core/).

## What's Changed
* small updates to the release doc by @chenmoneygithub in https://github.com/keras-team/keras-nlp/pull/1031
* Sampler docstring edit by @abuelnasr0 in https://github.com/keras-team/keras-nlp/pull/1033
* Fix program crash for id_to_token() method in SentencePieceTokenizer by @abuelnasr0 in https://github.com/keras-team/keras-nlp/pull/1040
* Update our release process to preview docs before release by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1043
* Add Whisper Tokenizer and Audio Feature Extractor by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/847
* Also strip padding token for opt by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1028
* Add regex dep by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1044
* Add BartSeq2SeqLM and conditional text generation with BART by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/974
* Support list/tuple inputs for special tokens in StartEndPacker layer by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/1045
* Support list/tuple inputs for special tokens in MultiSegmentPacker layer by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/1046
* Fix a misleading part of our cached MHA docs by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1048
* Always pass weight name by kwarg by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1053
* Always pass metrics in a list or dict by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1054
* Move `Defaults to` to end of arg docstring and standardise values by @SamuelMarks in https://github.com/keras-team/keras-nlp/pull/1057
* Fix beam search for BART by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/1058
* Replace tf.dtype with "dtype" by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1059
* Test shapes directly by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1064
* Clean up metrics tests by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1063
* Remove metrics merge tests by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1065
* Fix whisper feature inputs by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1069
* Always specify shape when creating variables by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1067
* Remove ragged support from position embeddings by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1068
* Clean up dtype handling for preprocessing layers by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1066
* Add BART finetuned on CNN+DM for summarisation by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/1060
* Fix saving bug by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1073
* Fix t5 forward pass by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1082
* Feat/make transformer decoder callable without causal mask by @ferraric in https://github.com/keras-team/keras-nlp/pull/1083
* Adding `GPTNeoXBackbone` by @shivance in https://github.com/keras-team/keras-nlp/pull/1056
* Add a common test case by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1095
* Update register_keras_serializable to use saving module by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1094
* Don't test tf format by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1104
* Add `GPTNeoXPreprocessor` by @shivance in https://github.com/keras-team/keras-nlp/pull/1093
* Split layers into layers/modeling & layers/preprocessing by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1102
* Fix merge conflict from #1102 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1105
* Add a common base class for generative models by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1096
* Add `GPTNeoXCausalLMPreprocessor` by @shivance in https://github.com/keras-team/keras-nlp/pull/1106
* Add Whisper Presets by @abheesht17 in https://github.com/keras-team/keras-nlp/pull/1089
* Refactor `RotaryEmbedding` and `GPTNeoXAttention` by @shivance in https://github.com/keras-team/keras-nlp/pull/1101
* Remove all the secret keys for ci by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1126
* Fix publish to pypi action by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1127
* Unexport models that are not in the 0.6 release by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1125
* Bump the version to 0.6.0 by @mattdangerw in https://github.com/keras-team/keras-nlp/pull/1128

## New Contributors
* @SamuelMarks made their first contribution in https://github.com/keras-team/keras-nlp/pull/1057
* @ferraric made their first contribution in https://github.com/keras-team/keras-nlp/pull/1083

**Full Changelog**: https://github.com/keras-team/keras-nlp/compare/v0.5.2...v0.6.0

<hr /><em>This discussion was created from the release <a href='https://github.com/keras-team/keras-nlp/releases/tag/v0.6.0'>v0.6.0</a>.</em>
